### PR TITLE
ECS fixes

### PR DESCRIPTION
**Pull request by i0rek**
This PR fixes the remaining problems with ECS. There are a couple backward incompatible changes:

* the main module no longer accepts `subnet_ids`
* every `routing_table_id` template parameter becomes `PublicRouteTableID`
* the `ecs-existing-vpc` template gets a new template parameter `PrivateRouteTableID`

Because of these changes, after this PR `v0.5.0` will be released.

The last commit contains the generated code.

### CODE REVIEW

File: examples/hcp-ecs-demo/main.tf [hunk 1]
```diff
@@ -23,12 +23,14 @@ resource "hcp_hvn" "main" {
...
-  hvn             = hcp_hvn.main
-  vpc_id          = module.vpc.vpc_id
-  subnet_ids      = module.vpc.public_subnets
-  route_table_ids = module.vpc.public_route_table_ids
+  hvn    = hcp_hvn.main
+  vpc_id = module.vpc.vpc_id
+  route_table_ids = concat(
+    module.vpc.private_route_table_ids,
+    module.vpc.public_route_table_ids,
+  )
```
Comment (i0rek) [inline examples/hcp-ecs-demo/main.tf:33]: This is part 1 of the fix for ECS.

File: main.tf [hunk 2]
```diff
@@ -68,11 +63,10 @@ resource "aws_vpc_peering_connection_accepter" "peer" {
 }
 
 resource "hcp_hvn_route" "peering_route" {
-  count            = length(var.subnet_ids)
   depends_on       = [aws_vpc_peering_connection_accepter.peer]
   hvn_link         = var.hvn.self_link
-  hvn_route_id     = var.subnet_ids[count.index]
-  destination_cidr = data.aws_subnet.selected[count.index].cidr_block
+  hvn_route_id     = "${var.hvn.hvn_id}-peering-route"
+  destination_cidr = data.aws_vpc.selected.cidr_block
```
Comment (i0rek) [inline main.tf:69]: This is part 2 of the ECS fix.

File: hcp-ui-templates/ec2/main.tf [hunk 3]
```diff
@@ -46,11 +46,10 @@ resource "hcp_hvn" "main" {
 
 module "aws_hcp_consul" {
   source  = "hashicorp/hcp-consul/aws"
-  version = "~> 0.4.2"
+  version = "~> 0.5.0"
```
Comment (chapmanc) [inline hcp-ui-templates/ec2/main.tf:49]: I don't see this as a released version ?

**Review by chapmanc — APPROVED** [review]
LGTM

**Pull request merged**